packages/core/src/dialects/db2/query-generator-typescript.ts

```diff
@@ -29,32 +30,49 @@ export class Db2QueryGeneratorTypeScript extends AbstractQueryGenerator {
     ]);
   }
 
+  private _getConstraintType(type: ConstraintType): string {
+    switch (type) {
+      case 'CHECK':
+        return 'K';
+      case 'FOREIGN KEY':
+        return 'F';
+      case 'PRIMARY KEY':
+        return 'P';
+      case 'UNIQUE':
+        return 'U';
+      default:
+        throw new Error(`Constraint type ${type} is not supported`);
+    }
+  }
+
   showConstraintsQuery(tableName: TableNameOrModel, options?: ShowConstraintsQueryOptions) {
     const table = this.extractTableDetails(tableName);
 
     return joinSQLFragments([
-      'SELECT c.TABSCHEMA AS "constraintSchema",',
+      'SELECT TRIM(c.TABSCHEMA) AS "constraintSchema",',
```

Comment:
Any specific reason why you are adding `TRIM()` to the schemas and not the other values?

Reply:
The `TABSCHEMA` value seems to return a trailing blank space hence the reason for adding `TRIM()`